lib/ElementHandle.js

```diff
@@ -137,24 +151,64 @@ class ElementHandle extends JSHandle {
     return {x, y, width, height};
   }
 
+  /**
+   * @return {!Promise<?{x: number, y: number, width: number, height: number}>}
+   */
+  async _assertBoundingBox() {
+    const boundingBox = await this.boundingBox();
+    if (!boundingBox)
+      throw new Error('Node is not visible');
+    return boundingBox;
+  }
+
   /**
    *
    * @param {!Object=} options
    * @returns {!Promise<Object>}
    */
   async screenshot(options = {}) {
-    await this._scrollIntoViewIfNeeded();
+    let needsViewportReset = false;
+
+    let boundingBox = await this._assertBoundingBox();
+
+    const viewport = Object.assign({}, this._page.viewport());
+
+    const newViewport = {
```

Comment:
Its a nitpicky thing, but can you build this rect inside the if statement below it. Like
```js
if (boundingBox.width > viewport.width || boundingBox.height > viewport.height) {
   const newViewport = ...
```
It makes it clearer at a glance that we don't always set a new viewport